src/main/java/pro/sky/mikhaillukichevcollections/service/DepartmentServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package pro.sky.mikhaillukichevcollections.service;
+
+import org.springframework.stereotype.Service;
+import pro.sky.mikhaillukichevcollections.model.Employee;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class DepartmentServiceImpl implements DepartmentService{
+
+    private final EmployeeService employeeService;
+
+    public DepartmentServiceImpl(EmployeeService employeeService) {
+        this.employeeService = employeeService;
+    }
+
+    public List<Employee> getEmployeesByDepartment(Integer department) {
+        return employeeService.getEmployees().values().stream().filter(employee -> employee.getDepartment().equals(department)).collect(Collectors.toList());
+    }
+
+    public List<Employee> getAllEmployees() {
+        return employeeService.getEmployees().values().stream().sorted((e1, e2)->e1.getDepartment().compareTo(e2.getDepartment())).collect(Collectors.toList());
+    }
+
+    public Employee findLowestSalaryEmployeeInDepartment(int department) {
+        final List<Employee> employeesByDepartment = this.getEmployeesByDepartment(department);
+        final List<Integer> salaries = employeesByDepartment.stream().map(employee -> employee.getSalary()).collect(Collectors.toList());
+        Optional<Integer> minSalary = salaries.stream().min((a, b) -> a - b);
+        Optional<Employee> minSalaryEmployee = employeesByDepartment.stream().filter(employee -> employee.getSalary().equals(minSalary.get())).findAny();
+        if (minSalaryEmployee.isPresent()) {
+            return minSalaryEmployee.get();
+        } else {
+            throw new RuntimeException("Min salary employee is not found for department " + department);
+        }
```

Comment:
как-то сложно) можно в одну строку решить:
```suggestion
        return employeeService.getEmployees().values().stream()
            .filter(employee -> employee.getDepartment().equals(department))
            .min(Employee::getSalary)
            .orElseThrow(() -> throw new RuntimeException("Min salary employee is not found for department " + department))
```

Reply:
в итоге твой код работает в 3 раза медленнее, чем мог бы, потому что ты 3 раза используешь стрим вместо одного. Всегда надо стараться все операции делать в одном стриме

Reply:
Супер, спасибо! Я понимал, что мой код выглядит как костыль, но получше придумать не удалось =)